dask/array/core.py

```diff
@@ -1713,13 +1728,19 @@ def __iter__(self):
 
     __array_priority__ = 11  # higher than numpy.ndarray and numpy.matrix
 
-    def __array__(self, dtype=None, **kwargs):
+    def __array__(self, dtype=None, copy=None):
+        if copy is False:
```

Comment:
This should at least get a deprecation warning. We shouldn't start raising here without warning

Reply:
I disagree. The user was explicitly asking for something impossible, and giving them back anything but an exception was in outright disregard of their (again, explicit) demands - e.g. a bug.

Reply:
We are ignoring all keywords intentionally here, if we want to change that then this should be deprecated. The potential breakage this might cause is not acceptable to me for a bug that hasn't hurt anyone as far as we know.

Reply:
Pandas for example is going through here with copy=False here and there and this shouldn’t start raising immediately.  The new copy behavior with copy=False needs a deprecation first

Reply:
Changed to deprecation.

Reply:
Yes, I don't disagree with the general point and if we'd start over I'd be in favour of raising directly. But I don't want this to become a breaking change immediately for something that has been there since forever and isn't really hurting anyone